### DISCUSSION TITLE
How to make a Blazor WebAssembly app listen for http requests?

### DISCUSSION THREAD

**Discussion by stefanalexandru02** [dotnet/aspnetcore#25994] [General] [answered]
Is it possible to have an HTTP server running in a Blazor WebAssembly App (on the client)? The scenario would be the following:

I have a client app (Blazor WebAssembly) running on a device.
Some other devices would have to do GET/POST requests to this client app.
It does not need to be super complicated, a simple text response from the WASM app would do just fine. Is it possible to do something like this or should I try to find another approach?

**Comment by SteveSandersonMS** [comment, marked as answer]
No, sorry. Browsers don't provide a mechanism for listening for HTTP requests. This kind of arbitrary TCP listener is not permitted within the browser sandbox, and so it's not possible for Blazor WebAssembly to do that.
